src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -68,6 +69,30 @@ trait Reporting { this: Context =>
   def featureWarning(msg: => String, pos: SourcePosition = NoSourcePosition): Unit =
     reporter.report(new FeatureWarning(msg, pos))
 
+  private[this] var reportedFeatures = Set[Symbol]()
```

Comment:
This will make reportedFeatures part of every context (and there are LOTS of those). I think it would be better to make the var a part of the Reporter class.